components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticatorTest.java

```diff
@@ -0,0 +1,236 @@
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_SESSION_INDEX)).thenReturn("123456789");
+        when(mockedHttpSession.getAttribute(SSOConstants.NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpSession.getAttribute(SSOConstants.SP_NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpServletRequest.getSession()).thenReturn(mockedHttpSession);
+
+        PrintWriter out = new PrintWriter("logout-request.txt");
+        when(mockedLocoutHttpServletResponse.getWriter()).thenReturn(out);
+
+        samlssoAuthenticator.initiateLogoutRequest(mockedHttpServletRequest, mockedLocoutHttpServletResponse,
+                mockedAuthenticationContext);
+        out.flush();
+        String postPage = FileUtils.readFileToString(new File("logout-request.txt"), "UTF-8");
+        Assert.assertTrue(postPage.contains("SAMLRequest"));
+        Assert.assertTrue(postPage.contains("RelayState"));
+    }
+
+    @Test(priority = 6)
+    public void testInitiateRedirectLogoutRequest() throws LogoutFailedException, IOException {
+
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL,
+                "https://localhost:9443/samlsso");
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.REQUEST_METHOD,
+                SSOConstants.REDIRECT);
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED, "true");
+        when(mockedAuthenticationContext.getAuthenticatorProperties()).thenReturn(authenticatorProperties);
+
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_USERNAME)).thenReturn("SomeUser");
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_SESSION_INDEX)).thenReturn("123456789");
+        when(mockedHttpSession.getAttribute(SSOConstants.NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpSession.getAttribute(SSOConstants.SP_NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpServletRequest.getSession()).thenReturn(mockedHttpSession);
+
+        samlssoAuthenticator.initiateLogoutRequest(mockedHttpServletRequest, mockedLocoutHttpServletResponse,
+                mockedAuthenticationContext);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(mockedLocoutHttpServletResponse).sendRedirect(captor.capture());
+        Assert.assertTrue(captor.getValue().contains("SAMLRequest"));
+        Assert.assertTrue(captor.getValue().contains("RelayState"));
+    }
+
+    @Test(priority = 7)
+    public void testGetContextIdentifier() {
+
+        when(mockedHttpServletRequest.getParameter("sessionDataKey")).thenReturn("123456789");
+        Assert.assertEquals("123456789", samlssoAuthenticator.getContextIdentifier(mockedHttpServletRequest));
```

Comment:
This is writing two tests in one test method right ? What if the first one fails ? We will not be able to get an idea on what will happen to the next one. Shall we split this to two tests ? Or else we can use data holders